server/SCHEMAS/mongodb.js

```diff
@@ -38,104 +43,270 @@ const QandASchema = new Schema({
+}
+
+// return the id counts from the database
+let countQuery = async () => {
+  try { return await Count.findOne() }
+  catch (err) { return err }
+}
+
+// udpdate the counts database
+let updateCounts = async () => {
+  try {
+    let questions = await count()
+    let answers = await countAnswers()
+    let data = { questions, answers }
+    let checkCounts = await Count.findOne()
+
+    if (checkCounts === null) {
+      await Count.create(data).lean()
+    } else {
+      let id = checkCounts._id
+      await Count.findByIdAndUpdate(id, { questions: data.questions }).lean()
+      await Count.findByIdAndUpdate(id, { answers: data.answers }).lean()
+    }
+  }
+  catch (err) { return err }
+}
+
+// increment the id counts for the database
+let incrementCount = async (countName, num) => {
+  try {
+    let count = await Count.findOne()
+
+    if (count === null) {
+      await updateCounts()
+      count = await Count.findOne()
+    }
+
+    let id = count._id
+    let update = {}
+    update[countName] = num
+
+    await Count.findByIdAndUpdate(id, update).lean()
+  }
+  catch (err) { return err }
+}
+
+// insert a new document as a question
 let insert = async (data) => {
-  await QandA.create(data)
-    .then(() => {})
-    .catch((err) => console.log('err in insert', err.message))
+  try { await QandA.create(data).lean() }
+  catch (err) { return err }
 }
 
+// insert an answer into the provided question document
 let answerInsert = async (question_id, data) => {
-  // query the db for the question and needed info
-  let query = { question_id }
-  let question = await QandA.find(query)
-  let id = question[0]._id
-  let answers = question[0].answers
-
-  // setup the new answer data
-  let answer = new Answer({
-    answer_id: data.id,
-    answer_body: data.body,
-    answer_date: data.date_written,
-    answerer_name: data.answerer_name,
-    answerer_email: data.answerer_email,
-    answer_reported: data.reported,
-    answer_helpfulness: data.helpful,
-    answer_photos: []
-  })
-
-  // if not present, insert the answer to the array from the question
-  let isPresent = false
-  answers.forEach(answer => {
-    if (!isPresent) {
-      if (answer.answer_id === parseInt(data.id)) {
-        isPresent = true
+  try {
+    let query = { question_id }
+    let question = await QandA.find(query).lean()
+    let id = question[0]._id
+    let answers = question[0].answers
+
+    let answer = new Answer({
+      answer_id: data.id,
+      answer_body: data.body,
+      answer_date: data.date_written,
+      answerer_name: data.answerer_name,
+      answerer_email: data.answerer_email,
+      answer_reported: data.reported,
+      answer_helpfulness: data.helpful,
+      answer_photos: data.photos || []
+    })
+
+    let isPresent = false
+    answers.forEach(answer => {
+      if (!isPresent) {
+        if (answer.answer_id === parseInt(data.id)) {
+          isPresent = true
+        }
       }
-    }
-  })
+    })
```

Comment:
I bet you could use $addToSet for this, just something to think about:
https://www.mongodb.com/docs/manual/reference/operator/update/addToSet/